File: fix_all_jsx_issues.py
#!/usr/bin/env python3
import os
import re
import glob

def fix_jsx_issues(file_path):
    """Fix all JSX issues in a file"""
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        original_content = content
        
        # Fix empty fragments
        content = re.sub(r'<></>', '<>', content)
        
        # Fix multiple JSX elements at root level
        # Pattern: return (\n    <></>\n    <Element>
        content = re.sub(
            r'return \(\s*\n\s*<></>\s*\n\s*<([^>]+)>',
            r'return (\n    <>\n    <\1>',
            content,
            flags=re.MULTILINE
        )
        
        # Fix patterns where we have <></> followed by other elements
        content = re.sub(
            r'<></>\s*\n\s*<([^>]+)>',
            r'<>\n    <\1>',
            content,
            flags=re.MULTILINE
        )
        
        # Fix unclosed tags
        content = re.sub(r'<([^>]+)>\s*$', r'<\1></\1>', content, flags=re.MULTILINE)
        
        # Fix specific patterns that are common
        content = re.sub(r'(\s*)<([^>]+)>\s*\n\s*<([^>]+)>', r'\1<>\n\1<\2>\n\1<\3>', content, flags=re.MULTILINE)
        
        if content != original_content:
            with open(file_path, 'w', encoding='utf-8') as f:
                f.write(content)
            print(f"Fixed JSX issues in: {file_path}")
            return True
        return False
        
    except Exception as e:
        print(f"Error processing {file_path}: {e}")
        return False

def main():
    """Main function to fix JSX issues in all files"""
    print("Fixing all JSX issues...")
    
    # Find all TypeScript/JavaScript files
    patterns = ['**/*.tsx', '**/*.ts']
    files_to_fix = []
    
    for pattern in patterns:
        files_to_fix.extend(glob.glob(pattern, recursive=True))
    
    # Filter out node_modules and other directories
    files_to_fix = [f for f in files_to_fix if 'node_modules' not in f and '.git' not in f]
    
    print(f"Found {len(files_to_fix)} files to check")
    
    fixed_count = 0
    for file_path in files_to_fix:
        if fix_jsx_issues(file_path):
            fixed_count += 1
    
    print(f"Fixed {fixed_count} files")

if __name__ == "__main__":
    main()

File: fix-all-jsx-issues.js
#!/usr/bin/env node

const fs = require('fs');
const path = require('path');
const glob = require('glob');

// Function to completely fix JSX issues in a file
function fixAllJSXIssues(filePath) {
  try {
    let content = fs.readFileSync(filePath, 'utf8');
    let modified = false;
    
    // Remove any BOM or hidden characters
    content = content.replace(/^\uFEFF/, '');
    
    // Ensure proper line endings
    content = content.replace(/\r\n/g, '\n').replace(/\r/g, '\n');
    
    // Fix the file structure completely
    if (content.includes('export default function')) {
      // Extract the function name
      const functionMatch = content.match(/export default function (\w+)/);
      if (functionMatch) {
        const functionName = functionMatch[1];
        
        // Create a proper file structure
        let newContent = '';
        
        // Add 'use client' if not present
        if (!content.includes("'use client';")) {
          newContent += "'use client';\n";
        } else {
          newContent += "'use client';\n";
        }
        
        // Add React import
        newContent += "import React from 'react';\n";
        
        // Add other necessary imports
        if (content.includes('<Head>') && !content.includes("import Head from 'next/head'")) {
          newContent += "import Head from 'next/head';\n";
        }
        if (content.includes('<Link') && !content.includes("import Link from 'next/link'")) {
          newContent += "import Link from 'next/link';\n";
        }
        if (content.includes('ArrowRight') && !content.includes("import { ArrowRight }")) {
          newContent += "import { ArrowRight } from 'lucide-react';\n";
        }
        if (content.includes('Footer') && !content.includes("import Footer from")) {
          newContent += "import Footer from '../components/Footer';\n";
        }
        
        newContent += "\n";
        
        // Extract the function body
        const functionStart = content.indexOf('export default function');
        const functionBody = content.substring(functionStart);
        
        // Clean up the function body
        let cleanFunctionBody = functionBody
          .replace(/export default function \w+\(\)\s*{/, `export default function ${functionName}() {`)
          .replace(/\s*return\s*\(\s*/, '\n  return (\n    ')
          .replace(/\s*\);\s*}\s*$/, '\n  );\n}');
        
        // Fix JSX structure
        cleanFunctionBody = cleanFunctionBody
          .replace(/<div>\s*<\/>/g, '<div></div>')
          .replace(/<\/div>\s*<\/>/g, '</div>')
          .replace(/<>(\s*<div[^>]*>)/g, '$1')
          .replace(/(<\/div>)\s*<\/>/g, '$1');
        
        newContent += cleanFunctionBody;
        
        // Only replace if there are actual changes
        if (newContent !== content) {
          content = newContent;
          modified = true;
        }
      }
    }
    
    if (modified) {
      fs.writeFileSync(filePath, content, 'utf8');
      return true;
    }
    
    return false;
  } catch (error) {
    console.error(`Error processing ${filePath}:`, error.message);
    return false;
  }
}

// Find all TSX files in the app directory
const pattern = 'app/**/*.tsx';
const files = glob.sync(pattern);

console.log(`Found ${files.length} TSX files to process`);

let fixedCount = 0;
files.forEach(file => {
  if (fixAllJSXIssues(file)) {
    fixedCount++;
    console.log(`Fixed: ${file}`);
  }
});

console.log(`Fixed ${fixedCount} files`);